<launch>
  <arg name="CONTROLLER_TYPE" default="auto" />
  <arg name="DEV" default="/dev/input/js0" />
  <arg name="POSE_REMAP" default="joy_pose" />
  <arg name="FRAME_ID" default="map" />

  <arg name="publish_js" default="false" />
  <arg name="gui" default="true" />

  <node pkg="joy" type="joy_node" name="joy_driver" output="screen" >
    <param name="dev" type="string" value="$(arg DEV)" />
    <param name="deadzone" value="0.2" />
    <param name="autorepeat_rate" value="40" />
    <param name="coalesce_interval" value="0.025" />
  </node>
  <node pkg="jsk_teleop_joy" type="joy.py" name="joy" output="screen"
        clear_params="true">
    <param name="controller_type" value="$(arg CONTROLLER_TYPE)" />
    <rosparam subst_value="true">
      plugins:
        'Buttons':
          class: 'ButtonMarkersMenu'
          args:
            menu: "overlay_menu"
            type: 1
            lock_xy: True
            publish_pose: True
            pose: $(arg POSE_REMAP)
            name: "joy_marker_array"
            namespace: "cylinder_array"
            frame_id: $(arg FRAME_ID)
            follow_view: true
            marker_pose: "marker_pose"
            show_label: True
            circle_cmd: "BTN_EXECUTE"
            triangle_cmd: "PREVIEW"
    </rosparam>
  </node>

  <group if="$(arg publish_js)">
    <node name="pseudo_robot_state_publisher"
          pkg="robot_state_publisher" type="robot_state_publisher" >
      <remap from="joint_states" to="robot_js" />
      <rosparam>
        tf_prefix: "pseudo"
        publish_frequency: 2.0
      </rosparam>
    </node>
  </group>

  <group if="$(arg gui)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_teleop_joy)/launch/config/sample_marker_array.rviz">
    </node>
  </group>

</launch>
